assembly/src/parsers/serde/deserialization.rs

```diff
@@ -373,14 +373,50 @@ impl Deserializable for Instruction {
             OpCode::CDropW => Ok(Instruction::CDropW),
 
             // ----- input / output operations --------------------------------------------------------
-            OpCode::PushConstants => {
+            OpCode::PushU8 => Ok(Instruction::PushU8(bytes.read_u8()?)),
+            OpCode::PushU16 => Ok(Instruction::PushU16(bytes.read_u16()?)),
+            OpCode::PushU32 => Ok(Instruction::PushU32(bytes.read_u32()?)),
+            OpCode::PushFelt => Ok(Instruction::PushFelt(bytes.read_felt()?)),
+            OpCode::PushWord => {
+                let mut constants = Vec::new();
+                for _ in 0..4 {
+                    constants.push(bytes.read_felt()?);
+                }
+                Ok(Instruction::PushWord(constants.try_into().unwrap()))
+            }
+            OpCode::PushU8List => {
+                let mut constants = Vec::new();
+                let length = bytes.read_u8()?;
+                for _ in 0..length {
+                    constants.push(bytes.read_u8()?);
+                }
+                Ok(Instruction::PushU8List(constants))
```

Comment:
Here, and in other places, we should check if length is greater than 1 and smaller than or equal to `MAX_PUSH_INPUTS`.

Also, I wonder if there is a way to reduce the amount of code duplication here - but I don't have any suggestions.